Application/Frontend/pages/api/registration.ts

```diff
@@ -20,17 +20,27 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
       email,
       phone,
       createdAt
-    } = req.body; // Intaking the data that has been sent from the client-side
+    } = req.body;
 
-    let client: MongoClient | null = null; // We need to assign something to the client so TypeScript is aware that it can be null if the connection fails
+    let client: MongoClient | null = null;
 
     try {
       client = new MongoClient(getMongoDbUri());
       await client.connect();
       const db = client.db('Accord');
 
       const accountsCollection = db.collection("Accounts");
-      // Querying the database by the userName we received
+
+      // First, check if an account with the same 'id' or 'userName' already exists
```

Comment:
- Great change to ensure that there is no duplication in username and ensuring that an account for that particular id has not been created before.